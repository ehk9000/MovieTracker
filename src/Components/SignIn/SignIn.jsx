import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import postUser from '../../Util/ApiCalls/postUser';
import Card from '../Card/Card';
import Styles from './SignIn.scss';

class SignIn extends Component {
	constructor() {
		super();
		this.state = {
			email: '',
			password: ''
		};
	}

	handleChange = e => {
		this.setState({
			[e.target.name]: e.target.value
		});
	};

	handleSubmit = async e => {
		e.preventDefault();
		const url = 'http://localhost:3000/api/users/';
		const { email, password } = this.state;
		const body = { email, password };
		try {
			const postResponse = await postUser(url, body);
			await console.log(postResponse);
		} catch (error) {
			// create method that will handle error, add classes & error styling etc.
			console.log(error);
		}
	};

	render() {
		return (
			<section className="signIn-background">
				<div className="signIn">
					<form onSubmit={this.handleSubmit} className="signInSubmit">
						<h3>LogIn</h3>
						<label htmlFor="email-existing">Email</label>
						<input type="email" name="email" placeholder="Email" id="email-existing" className="email" onChange={this.handleChange} />
						<label htmlFor="password-existing">Password</label>
						<input
							type="password"
							name="password"
							placeholder="Password"
							id="password-existing"
							onChange={this.handleChange}
						/>
						<input type="submit" className="submitAccountBtn" />
						<NavLink to="/createAccount" className="createAccountLink">
						Create Account
					</NavLink>
					</form>
				</div>
			</section>
		);
	}
}

export default SignIn;
