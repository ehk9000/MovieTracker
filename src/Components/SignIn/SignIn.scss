.signIn {
  display: flex;
  justify-content: center;
}
.signIn-background {
  background-image: url('https://images.unsplash.com/photo-1458053688450-eef5d21d43b3?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1352&q=80');
  height:100vh;
}

.email {
  margin-left: 30px;
}

input {
  margin: 2px 0;
}

.signInSubmit {
  padding: 10px;
  border: 3px solid black;
  background: #6B6D76;
  opacity: .8;
  width:500px;
  flex-direction: column;
  margin-top: 20px;
}

.signInSubmit label {
  margin: 0 5%;
}

.signInSubmit input {
  height: 30px;
  width: 57%;
}

.submitAccountBtn {
  height: 30px;
  background-color: #000;
  color: white;
  margin: 10px;
  margin-left: 120px;
  font-size: 20px;
}

.createAccountLink {
  margin-left: 120px;
}

.createAccountLink:hover {
  color:#8B0000;
}

.error-message {
  display: inline;
  color: #8B0000;
  text-align: center;
}

