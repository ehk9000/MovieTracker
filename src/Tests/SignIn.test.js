import React from 'react';
import ReactDOM from 'react-dom';
import {
  shallow
} from 'enzyme';
import SignIn from '../Components/SignIn/SignIn';


describe('SignIn', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( < SignIn / > );
  });

  it('should mock snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

})