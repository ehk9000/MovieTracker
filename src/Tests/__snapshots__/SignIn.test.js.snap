// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should mock snapshot 1`] = `ShallowWrapper {}`;
